### PR TITLE
Adding n-gram generator module to generators, implements #279.

### PR DESCRIPTION
**Pull request by springlaughing**
refinery
- [ ] Tested by creator on refinery
- [x] Tested by reviewer on refinery
- [x] Ensured that output of brick conforms with refinery structure (to be checked by reviewer)

API
- [x] Tested by creator on localhost:8000/docs
- [x] Tested by reviewer on localhost:8000/docs

common code
- [x] Common code tested in notebook/ script by creator
- [x] Common code tested in notebook/ script by reviewer
- [x] Common code contains docstrings and type hints

additional points:
- [x] Docstring and README is existing
- [x] Import statements (in `__init__.py`)
- [ ] (If necessary) Added dependency to requirements.txt
- [ ] (If necessary) Added dependency to issue for refinery env [here](https://github.com/code-kern-ai/refinery/issues/166)
- [ ] Published brick to Strapi CMS (locally)

Testing procedure: 
When testing in refinery, please ensure that the output of the brick conforms with the structure of refinery. 
For `extraction` bricks, this would be a tuple like `("label", span_start, span_end)`.
For `classification` bricks, this would be a string representing a label.
For `generator` bricks, this would either be a float, interger, string, boolean or a list, depending on the situation.

When testing the bricks, try to avoid using only one source of data. Meaning that you should not only use the `clickbait` sample
project, but also different texts with longer or more complex strings. 

A small refinery example project with a variation of texts called `bricks-test-data-project.zip` can be found in the bricks repository.

### CODE REVIEW

File: generators/ngram/nltk_ngram_generator/config.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            "variables": {
+                "ATTRIBUTE": {
+                    "selectionType": SelectionType.CHOICE.value,
+                    "addInfo": [
+                        BricksVariableType.ATTRIBUTE.value,
+                        BricksVariableType.GENERIC_STRING.value,
+                    ],
+                }
+            },
+        },
```
Comment (LeonardPuettmannKern) [inline generators/ngram/nltk_ngram_generator/config.py:32]: This section is missing the `NGRAM_SIZE` as an input, I would suggest to use this:

```suggestion
integrator_inputs={
    "name": "nltk_ngram_generator",
    "refineryDataType": RefineryDataType.TEXT.value,
    "variables": {
        "ATTRIBUTE": {
            "selectionType": SelectionType.CHOICE.value,
            "description": "only text fields",
            "optional": "false",
            "addInfo": [
                BricksVariableType.ATTRIBUTE.value,
                BricksVariableType.GENERIC_STRING.value
            ]
        },
        "NGRAM_SIZE": {
            "selectionType": SelectionType.INTEGER.value,
            "defaultValue": 2,
            "optional": "false",
            "addInfo": [
                BricksVariableType.GENERIC_INT.value
            ]
        }
    }
}
```

**Review by LeonardPuettmannKern — CHANGES_REQUESTED** [review]
The code looks great, well done and thank you for the contribution. I have tested the all snippets and only would suggest one change for the `config.py`. :-)

**Comment by springlaughing** [comment]
Got it, thanks, glad to participate :)

**Comment by springlaughing** [comment]
Should I do anything else for the merging?

File: generators/ngram/nltk_ngram_generator/config.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from util.configs import build_generator_function_config
+from util.enums import State, RefineryDataType, BricksVariableType, SelectionType
+from . import nltk_ngram, INPUT_EXAMPLE
```
Comment (LeonardPuettmannKern) [inline generators/ngram/nltk_ngram_generator/config.py:3]: This should be `from . import nltk_ngram_generator, INPUT_EXAMPLE`

File: generators/ngram/nltk_ngram_generator/config.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+def get_config():
+    return build_generator_function_config(
+        function=nltk_ngram,
```
Comment (LeonardPuettmannKern) [inline generators/ngram/nltk_ngram_generator/config.py:8]: Rename `nltk_ngram` to `nltk_ngram_generator`

**Review by LeonardPuettmannKern — COMMENTED** [review]
Just some renaming then it should be good to go. :)

**Comment by LeonardPuettmannKern** [comment]
@springlaughing I noticed that a name in the `config.py` also need to be fixed. After that, I can approve and will merge it! Thank you.

**Comment by springlaughing** [comment]
Ah, got it. Done.

**Review by LeonardPuettmannKern — APPROVED** [review]
Looks good, thanks again @springlaughing!

**Comment by LeonardPuettmannKern** [comment]
Merge implementing #279.

**Pull request merged**